File: Project 6 - Ruby to Java Conversion/Proj6Ahmed/Visitor.java
//Faizan Ahmed
//Project 6 - Visitor Class
//Nov 12, 2015

import java.text.*;

public class Visitor{
	
	//defining instance variables
	private String name;
	private char gender;
	private int age;
	private String citizenship;
	public int currentRoom;
	
	//constructor defined
	public Visitor(String theName, char theGender, int theAge, 
		String theCitizenship){
			name = theName;
			gender = theGender;
			age = theAge;
			citizenship = theCitizenship;
			currentRoom = 0;	
		}
		
		public void nextRoom(){
			if(currentRoom <= 6){
				currentRoom++;
			}
		}
		
		//.format() has been imported on top of class
		public String nameCitizenship(){
			String newString = String.format("-20s %-15s", name, citizenship);
			return newString;
		}
		
		//toString method
		public String toString(){
			String newString = String.format("%-20s %-15s %d", name, citizenship, currentRoom);
			return newString;
		}
		
}

File: Project 6 - Ruby to Java Conversion/white-house-tour/Visitor.rb
class Visitor
  attr_reader :name, :gender, :age, 
    :nationality, :current_room

    def initialize the_name, the_gender, the_age, the_citizenship
    @name = the_name
    @gender = the_gender
    @age = the_age
    @nationality = the_citizenship
    @current_room = 0
  end

  def next_room
  	if @current_room <= 6
      @current_room += 1
    end
  end
    
  def to_s
    return "%-20s %-15s %d" %
      [@name, @citizenship, @current_room]
  end
  
  def name_citizenship
    return "%-20s %-15s" %
      [@name, @citizenship]  	
  end
	
end